project/profiles/capacity4more/behat/features/group_content_permissions.feature

```diff
@@ -14,109 +14,859 @@ Feature: Test creation of the content permissions.
-    When  I go to "node/add/event"
+  Scenario Outline: As admin, access create content form, without purl prefix.
+    Given I am logged in as user "<user>"
+    When  I go to "<path>"
     Then  I should not have access to the page
 
+    Examples:
+      | user        | path                 |
+      | mariecurie  | /node/add/photoalbum |
+      | mariecurie  | /node/add/photo |
+      | mariecurie  | /node/add/event |
+      | mariecurie  | /node/add/document |
+      | mariecurie  | /node/add/discussion |
+      | mariecurie  | /node/add/wiki-page |
+
+  @api
+  Scenario: Set "Pending" group state to 'Public'. Anonymous user asked to log in.
+    Given I am logged in as user "mariecurie"
+    When  I change access of group "Pending group" to "Public"
+    Then  I am an anonymous user
+    And   I go to "pending"
+    And   I should see "Please log in"
+
   @api
-  Scenario: Group non member should not have access to group 'create event' form.
-    Given I am logged in as user "badhairday"
-    When  I go to "movie-corner/node/add/event"
+  Scenario Outline: As non member, member, group admin, group owner and site admin, access create content form, with purl prefix of pending public group.
+    Given  I am logged in as user "<user>"
+    When  I go to "<path>"
     Then  I should not have access to the page
 
+    Examples:
+      | user        | path                 |
+      | badhairday  | /pending/node/add/photoalbum |
+      | isaacnewton | /pending/node/add/photoalbum |
+      | turing      | /pending/node/add/photoalbum |
+      | alfrednobel | /pending/node/add/photoalbum |
+      | badhairday  | /pending/node/add/photo |
+      | isaacnewton | /pending/node/add/photo |
+      | turing      | /pending/node/add/photo |
+      | alfrednobel | /pending/node/add/photo |
+      | badhairday  | /pending/node/add/event |
+      | isaacnewton | /pending/node/add/event |
+      | turing      | /pending/node/add/event |
+      | alfrednobel | /pending/node/add/event |
+      | badhairday  | /pending/node/add/document |
+      | isaacnewton | /pending/node/add/document |
+      | turing      | /pending/node/add/document |
+      | alfrednobel | /pending/node/add/document |
+      | badhairday  | /pending/node/add/discussion |
+      | isaacnewton | /pending/node/add/discussion |
+      | turing      | /pending/node/add/discussion |
+      | alfrednobel | /pending/node/add/discussion |
+      | badhairday  | /pending/node/add/wiki-page |
+      | isaacnewton | /pending/node/add/wiki-page |
+      | turing      | /pending/node/add/wiki-page |
+      | alfrednobel | /pending/node/add/wiki-page |
+      | mariecurie  |  /pending/node/add/photoalbum |
+      | mariecurie  |  /pending/node/add/photo |
+      | mariecurie  |  /pending/node/add/event |
+      | mariecurie  |  /pending/node/add/document |
+      | mariecurie  |  /pending/node/add/discussion |
+      | mariecurie  |  /pending/node/add/wiki-page |
+
```

Comment:
Even if news is not meant to be a Group Content Type, today a GM/GA/GO/SA is able to create news inside a group - this generates all the expected behaviour except the display of an item in the discussions overview. So to prevent this behaviour, I would add /pending/node/add/news to this scenario